lib/commands/leave.ts

```diff
@@ -42,7 +42,7 @@ export default class LeaveCmd extends Command {
 
 		If you don't specify a device hostname or IP, this command will automatically
 		scan the local network for balenaOS devices and prompt you to select one
-		from an interactive picker. This usually requires root privileges.
```

Comment:
We need to understand this better -- whether or not root privileges are required. I gather that `balena join` and `balena leave` use the same implementation as `balena scan`:
```
const { discover } = await import('balena-sync');
await discover.discoverLocalBalenaOsDevices();
```
```
const { forms } = await import('balena-sync');
hostnameOrIp = await forms.selectLocalBalenaOsDevice();
```
Where `selectLocalBalenaOsDevice` calls `discoverLocalBalenaOsDevices`:
https://github.com/balena-io-modules/balena-sync/blob/v11.0.2/lib/forms.coffee#L24

If it was the case that we could simply drop the requirement for sudo, it would be great as there are significant issues with `sudo balena scan`. The most recent is #2149 that basically describes how `sudo balena scan` fails on Linux because `balena` is not in the path for `sudo`, and the user himself offers to update the docs to add the requirement of editing the PATH-for-sudo with a `visudo` command that would then be an additional installation requirement of the CLI. It would be much nicer if we simply didn't need sudo...

A preliminary digging reveals #542 and #1714. By the way, re your question there:

> pdcastro I think we should close either this or #1714 as duplicate, since they are the same issue.

The two issues are closely related but I think not duplicates. #542 is about removing the need for superuser privileges (so the command runs as an unprivileged user). #1714 is about implementing dynamic privilege escalation, so the CLI dynamically raises a privilege popup (GUI) or password prompt on the command line. In both issues, `sudo` would be avoided, but by very different means. :-)

Quoting Hedss in #542:
> As part of the discovery of local devices, we've needed to ensure that the implementation of MDNS we use listens/sends on every available NIC. Whilst our fix to MDNS (https://github.com/balena-io-modules/multicast-dns/commit/a15c63464eb43e8925b187ed5cb9de6892e8aacc) works fine when there's no other implementation of MDNS/DNS-SD running, it requires `resin-cli` to be run using `sudo` when there *are* other instances (as `224.0.0.251:5353` needs to be shared). This is not ideal.

I haven't understood it to the point of knowing whether simply removing the sudo requirement would cause `balena scan` to fail frequently or rarely. If the situation was detectable at runtime, we could try to escalate privileges only when really required.

Reply:
@pdcastro I don't have time at the moment to do a full investigation of `balena scan`, so will move this to draft for now.

Reply:
@srlowe, we can get this PR merged (it's great to remove the `scandevices` internal command 👍 ) almost as it is (or even as it is...). How about just rewording the following:

Original: `This requires root privileges.`
Current PR: (deleted)
Proposed: `This may require administrator/root privileges.`

Then the investigation re removing sudo could be left for #542, #1714, #2149.

Reply:
@pdcastro Ok sure, sounds good.  I'd like to get the scan command investigated too, but I just have too much stuff open at the moment.